src/sentry/static/sentry/app/views/settings/organizationDeveloperSettings/sentryApplicationRow.jsx

```diff
@@ -20,13 +22,33 @@ export default class SentryApplicationRow extends React.PureComponent {
     installs: PropTypes.array,
     onInstall: PropTypes.func,
     onUninstall: PropTypes.func,
+    onRemoveApp: PropTypes.func,
     showPublishStatus: PropTypes.bool,
   };
 
   static defaultProps = {
     showPublishStatus: false,
   };
 
+  renderRemoveApp(app) {
+    const message = t(
+      `Are you sure you want to remove the ${app.slug} application ? \
+       If your application has any installations, data from those \
+       installations will be removed along with the installation itself. \
+       This is a permanent action.`
```

Comment:
Feels like this could be more concise. Maybe:

> Deleting ${app.slug} will also delete all of its installations. This action is permanent. Do you want to continue?

🤷‍♂️